src/pveclib/vec_int128_ppc.h

```diff
@@ -6848,4 +7005,140 @@ vec_vmsumoud (vui64_t a, vui64_t b, vui128_t c)
   return vec_adduqm (res, c);
 #endif
 }
+
+/** \brief Vector Shift Left Double Quadword by Bit Immediate.
+ *
+ *  Return a vector __int128 that is bits shb:shb+127
+ *  from the (256-bit) double quadword (vra || vrb).
+ *  The shift amount is constant immediate value in the range 0-7.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |   8   | 1 cycle  |
+ *  |power9   |  11   | 1/cycle  |
+ *
+ *  @param vra upper 128-bits of the 256-bit double quadword vector.
+ *  @param vrb lower 128-bits of the 256-bit double quadword vector.
+ *  @param shb Shift amount in the range 0-7.
+ *  @return 128-bits from bits shb:shb+127.
+ */
+static inline vui128_t
+vec_vsldbi (vui128_t vra, vui128_t vrb, const unsigned int shb)
+{
+  vui128_t result;
+
+  if (__builtin_constant_p (shb) && (shb < 8))
+    {
+#if defined (_ARCH_PWR10)  && (__GNUC__ >= 10)
+      __asm__(
+	  "vsldbi %0,%1,%2,%3;\n"
+	  : "=v" (result)
+	  : "v" (vra), "v" (vrb), "K" (shb)
+	  : );
+#else
+      /* For Power7/8/9 the quadword shift left/right instructions
+       * only handle 128-bits.
+       * So shift vra and vrb separately then combine those into
+       * a single 128-bit result.
+       */
+      if (shb > 0)
+	{
+	  const vui8_t vshl = vec_splat_u8 (shb);
+	  const vui8_t vshr = vec_splat_u8 (8 - shb);
+	  const vui8_t zero = vec_splat_u8 (0);
+	  vui8_t lowbits, highbits;
+
+	  /* Left double shift by 1 octet 'zero' || vrb to isolate the
```

Comment:
"Left double shift" ... maybe rewrite to just "Left shift". (You use "right shift" later, anyway.)

Reply:
Yes but the sequence starts with vec_sld (Vector Shift Left Double by Octet Immediate).
The left double is important here because we are crossing registers.

No change for now.

Reply:
How about change to:

/* Shift left double by 15 octet vra || 'zero' to isolate the

Reply:
hmm. so the `vra || 'zero'` is supposed to represent the double quadword, correct?  Maybe put that next to that noun, then:
> Shift double quadword (vra || 'zero') left by 15 octets

?